packages/tofu/src/util.js

```diff
@@ -178,10 +185,20 @@ function getParents(nodePath) {
   return parents
 }
 
-function isInFunctionDeclaration(path) {
-  return getParents(path.parentPath).some(
-    (parent) =>
-      parent.type === 'FunctionDeclaration' ||
-      parent.type === 'FunctionExpression'
-  )
+/**
+ * Determines if this `Node` is a descendant of a `FunctionDeclaration` or `FunctionExpression`.
+ *
+ * @param {import("@babel/traverse").NodePath} nodePath
+ * @returns {boolean}
+ */
+function isInFunctionDeclaration(nodePath) {
+  let target = nodePath.parentPath
+  while (target) {
+    // if function declaration found, short-circuit instead of continuing
+    if (target.isFunctionDeclaration() || target.isFunctionExpression()) {
+      return true
+    }
+    target = target.parentPath
+  }
+  return false
```

Comment:
This is opportunistic and I changed it while debugging.  This is more performant than the previous implementation, but not by an order of magnitude.

Reply:
cc @kumavis